src/vl-map-action.js

```diff
@@ -104,34 +101,36 @@ export class VlMapAction extends vlElement(HTMLElement) {
   }
 
   _layerChangedCallback() {
-    this._mapElement.ready.then(() => this._processAction(this.layer));
+    if (this._layerElement) {
+      this.layer = this._layerElement.layer;
+    }
   }
 
   _createAction() {
     console.warn('implementatie van _createAction ontbreekt');
   }
 
-  _processAction(layer) {
-    if (layer) {
-      this._action = this._createAction(layer);
+  _processAction() {
+    this._mapElement.ready.then(() => {
       if (this._action) {
-        this._mapElement.addAction(this._action);
-        if (this._active) {
-          this.activate();
+        this._mapElement.removeAction(this._action);
+      }
+
+      if (this.layer) {
+        this._action = this._createAction(this.layer);
+        if (this._action) {
+          this._mapElement.addAction(this._action);
+          if (this._active) {
+            this.activate();
+          }
```

Comment:
gaat ge dit dan bij elke actie opnieuw moeten implementeren? da lijkt me tricky om ne keer te vergeten ... 
geen nood aan een baseclass voor acties?

Reply:
Dit is de base class, dus moet niet bij elke actie voorzien worden.

Reply:
ah just, ik was aan het verwarren dat dit de "teken"-actie was. 
hmm, maar er gaan sowieso wel acties zijn die (potentieel) met meer dan 1 laag werken, dus dan weet ik niet of het een geweldig slim idee is om "layer" en niet "layerS" te voorzien in de base class?

Reply:
Viel me inderdaad ook op, maar was het blijkbaar vergeten te zeggen in een comment. Ik zou er ook wel layers van maken.

Reply:
Het achterliggend vl-mapactions framework heeft bij de kaart acties maar ondersteuning voor 1 layer. Hierdoor heb ik het vermoeden dat een kaart actie in het verleden en daarom vermoedelijk ook in de toekomst per kaartlaag voorzien zal worden.

Er is één uitzondering, namelijk de VlSelectActions class die blijkbaar met een layer configuratie object werkt. Ik zou eerder deze als uitzondering behandelen dan layers als default op de base class te zetten.

Reply:
In de OpenLayers API (https://openlayers.org/en/latest/apidoc/module-ol_interaction_Interaction-Interaction.html) is er nog geen sprake van layers bij een (inter)actie, dus misschien moeten we ook maar beter die weg op gaan en het in de subclasses steken?

Reply:
Als ge in de superclass "layer" gaat steken, en dan in SelectAction "layers", vind ik dat een subclass violation ...

Reply:
Die weg kan bewandeld worden, al lijkt me dat op dit moment niet nodig. Ik zou VlSelectActions niet laten overerven van een action class, aangezien het eigenlijk een verzameling van actions is, in die zin is er geen subclass violation.

Reply:
export class VlSelectAction extends VlMapAction
export class VlSelectActions extends VlSelectAction

dus waarom zou dat geen overerving zijn?

Reply:
Er is dus al een subclass violation op vl-mapactions niveau, maar die moet daarom niet overgenomen worden op webcomponent niveau.

Reply:
en als ge ne keer kijkt naar de API van OpenLayers zijn er ook (inter)acties zonder layer (bvb https://openlayers.org/en/latest/apidoc/module-ol_interaction_DragRotate-DragRotate.html), die mij eerder zelfs in de richting lijken te duwen van "bepaal het in de subclasses" ...

Reply:
Dat zijn (inter)acties die op een ander niveau, die van de map, gedeclareerd worden via attributen: data-vl-disable-rotation is hier een voorbeeld van dat op dit moment ook al geïmplementeerd werd. Op dit moment is er een onderscheid tussen OL acties en vl-mapaction acties.

Reply:
de vraag is, of dat dan OK is en we niet zouden 'standing on the shoulders of giants' en de API van de webcomponenten toch wat de API van OL te laten volgen?

Reply:
Omgekeerde vraagstelling kan ook, moeten we voor elke OL actie een webcomponent wrapper voorzien en/of rekening houden dat die er potentieel ooit komt of beperken we ons voorlopig tot die van de vl-mapactions die tot op heden voldoende bleken te zijn in de kaart intensieve applicaties?

Duidelijk iets dat op niveau van de technical leads besproken en beslist kan worden. Tot dan unresolved en on hold.

Reply:
Ik zou er alleszins rekening mee houden dat iemand de functionele vraag gaat stellen om iets te gebruiken wat in OL zit ja ...

Reply:
Als dat voor jou betekent dat de layer getter en setter gedupliceerd moet worden op de implementatie classes, zal ik dat uitvoeren.

Reply:
Absoluut. Het is alleen op het echt in de "absolute base class" te steken, dat ik wat gevaarlijk vond.

Reply:
Ik beperk me vaak tot wat op dit moment nodig is, maar wel rekening houdend met wijzigingen en uitbreidingen in de toekomst. Deze uitbreiding is bijvoorbeeld ook perfect mogelijk om maar te introduceren wanneer het nodig blijkt, zelfs zonder breaking changes. Aangezien er bezorgdheid is en de inspanning beperkt is, kan het uiteraard nu ook al uitgevoerd worden. Ik zal dat doen.